device/tt_cluster_descriptor.cpp

```diff
@@ -437,44 +437,31 @@ std::unique_ptr<tt_ClusterDescriptor> tt_ClusterDescriptor::create_from_yaml(
+    const std::vector<chip_id_t> &logical_device_ids, tt::ARCH arch) {
     std::unique_ptr<tt_ClusterDescriptor> desc = std::unique_ptr<tt_ClusterDescriptor>(new tt_ClusterDescriptor());
-
-    // Some users need not care about physical ids, can provide empty set.
-    auto use_physical_ids = physical_mmio_device_ids.size() ? true : false;
-    auto largest_workload_logical_device_id = *logical_mmio_device_ids.rbegin();  // Last element in ordered set.
-    auto num_available_physical_devices = physical_mmio_device_ids.size();
-    auto required_physical_devices = largest_workload_logical_device_id + 1;
-
-    log_debug(
-        tt::LogSiliconDriver,
-        "{} - use_physical_ids: {} largest_workload_logical_device_id: {} num_available_physical_devices: {} "
-        "required_physical_devices: {}",
-        __FUNCTION__,
-        use_physical_ids,
-        largest_workload_logical_device_id,
-        num_available_physical_devices,
-        required_physical_devices);
-
-    log_assert(
-        !use_physical_ids || num_available_physical_devices >= required_physical_devices,
-        "Insufficient silicon devices. Workload requires device_id: {} (ie. {} devices) but only {} present",
-        largest_workload_logical_device_id,
-        required_physical_devices,
-        num_available_physical_devices);
-
-    // All Grayskull devices are MMIO mapped so physical_mmio_device_ids correspond to all available devices
-    for (auto &logical_id : logical_mmio_device_ids) {
-        auto physical_id = use_physical_ids ? physical_mmio_device_ids.at(logical_id) : -1;
-        desc->chips_with_mmio.insert({logical_id, physical_id});
+    
+    BoardType board_type;
+    switch(arch){
+        case tt::ARCH::WORMHOLE_B0:
+            board_type = BoardType::N150;
+            break;
+        case tt::ARCH::BLACKHOLE:
+            board_type = BoardType::P150A;
+            break;
+        default:
+            log_error("Unsupported architecture for mock cluster");
+            break;
+    }
+
+    for (auto &logical_id : logical_device_ids) {
         desc->all_chips.insert(logical_id);
-        eth_coord_t chip_location{logical_id, 0, 0, 0};
+        eth_coord_t chip_location{0, logical_id, 0, 0, 0};
         desc->chip_locations.insert({logical_id, chip_location});
-        desc->coords_to_chip_ids[chip_location.rack][chip_location.shelf][chip_location.y][chip_location.x] =
-            logical_id;
-        log_debug(
-            tt::LogSiliconDriver, "{} - adding logical: {} => physical: {}", __FUNCTION__, logical_id, physical_id);
+        desc->coords_to_chip_ids[chip_location.rack][chip_location.shelf][chip_location.y][chip_location.x] = logical_id;
+        log_debug(tt::LogSiliconDriver, "{} - adding logical: {}", __FUNCTION__, logical_id);
+        desc->chip_board_type.insert({logical_id, board_type});
+        desc->chips_with_mmio.insert({logical_id, logical_id});
```

Comment:
@vtangTT is mmio required by simulator?
Would we ever want not to have mmio chips in this mock cluster descriptor, or we're always fine with having mmio?
@mbezuljTT Is this whole create_mock_cluster in line with your expectations for mock class?

Reply:
mmio is not required by the simulator however, the `chip_with_mmio` map has to be populated or else the metal cluster will segfault